File: python/src/circle_ai/safety_child/safety_child_domain_context.py
# safety_child_domain_context.py
#
# Port of CircleAI.Safety.Child SafetyChildDomainContext.cs (C# — the EXACT spec).
#
# Static domain-context data for the child-safety / safeguarding vertical: the
# system-prompt snippet, compliance flags and suggested tools. (C# namespace is
# CircleAI.SafetyChild.) The C# `static class` with get-only static properties
# maps to a Python class with immutable class attributes.

from __future__ import annotations

from typing import Sequence


class SafetyChildDomainContext:
    """Domain context for the Child Safety vertical (mirrors
    ``CircleAI.SafetyChild.SafetyChildDomainContext``).
    """

    SystemPromptSnippet: str = (
        "[DOMAIN: Safety.Child] Child safety and safeguarding assistant for "
        "parents and educators. Help with online safety education, age-appropriate "
        "device rules, recognising grooming signs, reporting abuse, and digital "
        "literacy. Always prioritise child welfare. IMPORTANT: For immediate child "
        "safety concerns, contact SAPS (10111) or Childline (116). Compliance: "
        "Children's Act 38/2005, POPIA (children's data), FILMS_PUBLICATIONS_ACT, "
        "Cybercrimes Act."
    )

    ComplianceFlags: Sequence[str] = (
        "Childrens_Act_38_2005",
        "POPIA_Children",
        "Films_Publications_Act",
        "Cybercrimes_Act",
        "Emergency_116",
    )

    SuggestedTools: Sequence[str] = (
        "parental_controls",
        "web_search",
        "document_editor",
        "reporting_tools",
    )

File: rust/src/safety_child/safety_child_domain_context.rs
//! safety_child_domain_context.rs
//!
//! Rust port of `src/CircleAI.Safety.Child/SafetyChildDomainContext.cs`
//! (C# namespace `CircleAI.SafetyChild`) — the static domain descriptor for the
//! child-safety / safeguarding vertical. Strings are reproduced byte-for-byte.

/// (Safety.Child) Static domain context for the child-safeguarding vertical.
///
/// Mirrors `static class SafetyChildDomainContext`.
pub struct SafetyChildDomainContext;

impl SafetyChildDomainContext {
    /// The system-prompt snippet injected ahead of child-safety turns.
    pub const SYSTEM_PROMPT_SNIPPET: &'static str = "[DOMAIN: Safety.Child] Child safety and safeguarding assistant for parents and educators. Help with online safety education, age-appropriate device rules, recognising grooming signs, reporting abuse, and digital literacy. Always prioritise child welfare. IMPORTANT: For immediate child safety concerns, contact SAPS (10111) or Childline (116). Compliance: Children's Act 38/2005, POPIA (children's data), FILMS_PUBLICATIONS_ACT, Cybercrimes Act.";

    /// Compliance flags applicable to this vertical.
    pub fn compliance_flags() -> Vec<String> {
        vec![
            "Childrens_Act_38_2005".to_string(),
            "POPIA_Children".to_string(),
            "Films_Publications_Act".to_string(),
            "Cybercrimes_Act".to_string(),
            "Emergency_116".to_string(),
        ]
    }

    /// Tools suggested for this vertical.
    pub fn suggested_tools() -> Vec<String> {
        vec![
            "parental_controls".to_string(),
            "web_search".to_string(),
            "document_editor".to_string(),
            "reporting_tools".to_string(),
        ]
    }
}
